compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt

```diff
@@ -229,38 +233,42 @@ private class AnimatedMouseWheelScrollNode(
         if (target.isLowScrollingDelta()) {
             return
         }
-        scrollableState.scroll {
-            var requiredAnimation = true
-            var lastValue = 0f
-            val anim = AnimationState(0f)
-            while (requiredAnimation) {
-                requiredAnimation = false
-                val durationMillis = (abs(target - anim.value) / speed)
-                    .roundToInt()
-                    .coerceAtMost(maxDurationMillis)
-                anim.animateTo(
-                    target,
-                    animationSpec = tween(
-                        durationMillis = durationMillis,
-                        easing = LinearEasing
-                    ),
-                    sequentialAnimation = true
-                ) {
-                    val delta = value - lastValue
-                    if (!delta.isLowScrollingDelta()) {
-                        val consumedDelta = scrollBy(delta)
-                        if (!(delta - consumedDelta).isLowScrollingDelta()) {
+        var requiredAnimation = true
+        var lastValue = 0f
+        val anim = AnimationState(0f)
+        while (requiredAnimation) {
```

Comment:
Shouldn't we keep `while` inside scroll? To cancel our mouse wheel animation completely, when something starts to scroll the list programmatically

Reply:
It was my first try, but no - this way we lose not consumed delta

Reply:
Semantically, we need to stop the animation. Can you explain the issue more? May be we can stop the animation, but fix the issue another way.

Reply:
> Semantically, we need to stop the animation. 

Discussed that if we stop the animations soon, we'll see sharp unsmooth scroolling (if there are 2 scrolls involved)

Reply:
> Semantically, we need to stop the animation

No, it's about finalizing of already received user input that should be finished. The animation's inputs are dynamic, so it's about finishing, not repeating